modules/mobile/spec/request/facility_eligibility_request_spec.rb

```diff
@@ -0,0 +1,158 @@
+          expect(response.parsed_body['meta']).to eq(
+            { 'pagination' => { 'currentPage' => 1, 'perPage' => 3,
+                                'totalPages' => 2, 'totalEntries' => 4 } }
+          )
+          expect(response.parsed_body['links']).to eq(
+            {
+              'self' => 'http://www.example.com/mobile/v0/appointments/facility/eligibility?page[size]=3&page[number]=1&serviceType=primaryCare&facilityIds[]=100&facilityIds[]=101&facilityIds[]=102&facilityIds[]=103&type=request',
+              'first' => 'http://www.example.com/mobile/v0/appointments/facility/eligibility?page[size]=3&page[number]=1&serviceType=primaryCare&facilityIds[]=100&facilityIds[]=101&facilityIds[]=102&facilityIds[]=103&type=request',
+              'prev' => nil,
+              'next' => 'http://www.example.com/mobile/v0/appointments/facility/eligibility?page[size]=3&page[number]=2&serviceType=primaryCare&facilityIds[]=100&facilityIds[]=101&facilityIds[]=102&facilityIds[]=103&type=request',
+              'last' => 'http://www.example.com/mobile/v0/appointments/facility/eligibility?page[size]=3&page[number]=2&serviceType=primaryCare&facilityIds[]=100&facilityIds[]=101&facilityIds[]=102&facilityIds[]=103&type=request'
+            }
+          )
+        end
+      end
+    end
+
+    context 'Invalid parameters' do
+      context 'when invalid serviceType is provided' do
+        let(:params) do
+          { serviceType: 'notatype', facilityIds: ['942'], type: 'request' }
+        end
+
+        before do
+          get '/mobile/v0/appointments/facility/eligibility', params: params, headers: iam_headers
+        end
+
+        it 'returns 400 with an error message' do
+          expect(response).to have_http_status(:bad_request)
+        end
+
+        it 'returns expected error message' do
+          expect(response.parsed_body.dig('errors', 0,
+                                          'detail')).to eq('"notatype" is not a valid value for "serviceType"')
+        end
+      end
+
+      context 'bad facility id' do
+        let(:params) do
+          { serviceType: 'primaryCare', facilityIds: ['1234567'], type: 'request' }
+        end
+
+        before do
+          VCR.use_cassette('facility_eligibility/get_patient_appointment_metadata_bad_facility',
+                           match_requests_on: %i[method uri]) do
+            get '/mobile/v0/appointments/facility/eligibility', params: params, headers: iam_headers
+          end
+        end
+
+        it 'returns 200' do
+          expect(response).to have_http_status(:ok)
+        end
+
+        it 'matches schema' do
+          expect(response.body).to match_json_schema('facility_eligibility')
+        end
+
+        it 'assumes facility exists but does not support requests' do
+          expect(response.parsed_body.dig('data', 0, 'attributes', 'reason'))
+            .to eq('appointment requests are disabled for the clinical service at the facility')
```

Comment:
Interesting. Did you record the cassette for this by going to staging or a review instance?

Reply:
Went into staging and called the upstream service directly